include/ring_buffer.h

```diff
@@ -0,0 +1,91 @@
+/**
```

Comment:
Just saw this - can we put ring_buffer into it's own directory  or an existing one, like util?